components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2106,4 +2282,176 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
+     * @param authenticationMethod authentication methid used to authenticate to the token endpoint
+     * @throws IdentityOAuthClientException
+     */
+    private void filterAllowedFAPITokenAuthMethods(String authenticationMethod) throws IdentityOAuthClientException {
+
+        List<String> allowedAuthMethods = IdentityUtil.getPropertyAsList(FAPI_CLIENT_AUTH_METHOD_CONFIGURATION);
+        if (allowedAuthMethods.isEmpty()) {
+            allowedAuthMethods.add(OAuthConstants.PRIVATE_KEY_JWT);
+            allowedAuthMethods.add(OAuthConstants.TLS_CLIENT_AUTH);
+        }
+        if (authenticationMethod != null && !allowedAuthMethods.contains(authenticationMethod)) {
+            throw handleClientError(INVALID_REQUEST, "Invalid token endpoint authentication method requested.");
+        }
+    }
+
+    /**
+     * FAPI validation to restrict the signature algorithms.
+     * Link - https://openid.net/specs/openid-financial-api-part-2-1_0.html#algorithm-considerations
+     * @param signatureAlgorithm signature algorithm used to sign the assertions.
+     * @throws IdentityOAuthClientException
+     */
+    private void filterAllowedFAPISignatureAlgorithms(String signatureAlgorithm)
+            throws IdentityOAuthClientException {
+
+        List<String> allowedSignatureAlgorithms = IdentityUtil
+                .getPropertyAsList(FAPI_SIGNATURE_ALGORITHM_CONFIGURATION);
+        if (allowedSignatureAlgorithms.isEmpty()) {
+            allowedSignatureAlgorithms.add(OAuthConstants.SignatureAlgorithms.ES256);
+            allowedSignatureAlgorithms.add(OAuthConstants.SignatureAlgorithms.PS256);
+        }
+        if (signatureAlgorithm != null && !allowedSignatureAlgorithms.contains(signatureAlgorithm)) {
+            throw handleClientError(INVALID_REQUEST, "Invalid signature algorithm requested");
+        }
+    }
+
+    /**
+     * FAPI validation to restrict the encryption algorithms.
+     * Link - https://openid.net/specs/openid-financial-api-part-2-1_0.html#encryption-algorithm-considerations
+     * @param encryptionAlgorithm
+     * @throws IdentityOAuthClientException
+     */
+    private void filterAllowedFAPIEncryptionAlgorithms(String encryptionAlgorithm)
+            throws IdentityOAuthClientException {
+
+        if (encryptionAlgorithm.equals(OAuthConstants.RESTRICTED_ENCRYPTION_ALGORITHM)) {
+            throw handleClientError(INVALID_REQUEST, "Invalid encryption algorithm requested");
+        }
+    }
+
+
+    /**
+     * If there are multiple hostnames in the registered redirect_uris,
+     * the Client MUST register a sector_identifier_uri.
+     * https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg
+     * @param redirectURIs list of callback urls sent in the request
+     * @throws IdentityOAuthClientException
+     */
+    private void validateRedirectURIForPPID(List<String> redirectURIs) throws IdentityOAuthClientException {
```

Comment:
Can't see the validation checking if a sector_identifier_uri is present. If the validation is throwing an error mentioning sector_identifier_uri is not available then the sector_identifier_uri check should also be in the method.

Reply:
Otherwise another way to handle this is to check and get if multiple hostnames are used. and then if a valid sector uri is not provided then throw the exception

Reply:
chaned in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/feb4731bf60075ddf78c6651b076e544ceea8add